test/libsolidity/semanticTests/constants/function_unreferenced.sol

```diff
@@ -0,0 +1,11 @@
+contract B {
+  function g() public {}
+}
+contract C is B {
+  bytes4 constant s2 = B.g.selector;
+  function f() external pure returns (bytes4) { return s2; }
+}
```

Comment:
I'd add more cases from #10891, maybe combined into a single test:

```solidity
contract B {
    function ext() external {}
    function pub() public {}
}

contract C is B {
    function test() public {
        B.ext.selector;
        B.pub.selector;
        this.ext.selector;
        pub.selector;
    }
}

contract D {
    function test() public {
        B.ext.selector;
        B.pub.selector;
    }
}
```

Reply:
The problem is that those cannot be done as a semantic test.

Reply:
Perhaps split into two tests

```
contract B {
    function ext() external {}
    function pub() public {}
}

contract C is B {
    function test() public returns (bytes4, bytes4, bytes4, bytes4) {
        return (B.ext.selector, B.pub.selector, this.ext.selector, pub.selector);
    }
}
// ====
// compileViaYul: true
// ----
// test() ->
```

and

```
contract B {
    function ext() external {}
    function pub() public {}
}

contract D {
    function test() public returns (bytes4, bytes4) {
        return (B.ext.selector, B.pub.selector);
    }
}
// ====
// compileViaYul: true
// ----
// test() -> 0xcf9f23b500000000000000000000000000000000000000000000000000000000, 0x7defb41000000000000000000000000000000000000000000000000000000000
```

Reply:
Does the tuple really trigger the same code path? I'll add a syntax test for the others just to be sure.

Reply:
> Does the tuple really trigger the same code path? I'll add a syntax test for the others just to be sure.

Yes, just verified that it does.